### PR TITLE
[BRE-53] publishing workflow

### PR DESCRIPTION
**Pull request by michalchecinski**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/BRE-53

## 📔 Objective

Add cd workflow to publish to pypi.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+          python-version-file: ".python-version"
+
+      - name: Install hatch
+        run: pip install hatch
+
+      - name: Get current version
+        run: echo "OLD_VERSION=$(hatch version)" >> $GITHUB_ENV
+
+      - name: Bump version
+        run: hatch bump ${{ needs.version-type.outputs.type }}
```
Comment (mimartin12) [inline .github/workflows/cd.yml:46]: ⚠️ This looks like it's calling the incorrect output, by the looks of `_version_type`.
```yaml
on:
  workflow_call:
    outputs:
      version:
        description: "version to be built"
        value: ${{ jobs.version.outputs.bump_type }}
```

it should be:
```suggestion
        run: hatch bump ${{ needs.version-type.outputs.version }}
```

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+      - name: Set up Python
+        uses: actions/setup-python@0a5c61591373683505ea898e09a3ea4f39ef2b9c # v5.0.0
+        with:
+          python-version-file: ".python-version"
+
+      - name: Install hatch
+        run: pip install hatch
+
+      - name: Login to Azure
```
Comment (mimartin12) [inline .github/workflows/cd.yml:126]: ⛏️  Consistent with other uses from above.
```suggestion
      - name: Login to Azure - CI Subscription
```

File: .github/workflows/enforce-labels.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+name: Enforce PR labels
+
+on:
+  pull_request:
+    types: [labeled, unlabeled, opened, reopened, synchronize]
+jobs:
+  enforce-labels:
+    name: Enforce labels
+    runs-on: ubuntu-latest
```
Comment (mimartin12) [inline .github/workflows/enforce-labels.yml:10]:
```suggestion
    runs-on: ubuntu-22.04
```
❓  Should this be pinned?

**Review by mimartin12 — CHANGES_REQUESTED** [review]

File: .github/workflows/enforce-labels.yml (hunk 3 above)
Comment (michalchecinski) [reply .github/workflows/enforce-labels.yml:10]: It should. Thanks!

**Review by mimartin12 — APPROVED** [review]

**Review by mimartin12 — APPROVED** [review]

**Review by mimartin12 — APPROVED** [review]

**Pull request merged**